src/spec/cli.cr

```diff
@@ -0,0 +1,128 @@
+require "option_parser"
+
+# This file is included in the compiler to add usage instructions for the
+# spec runner on `crystal spec --help`.
+
+module Spec
+  # :nodoc:
+  class_property? use_colors = true
+
+  # :nodoc:
+  class_property pattern : Regex?
+
+  # :nodoc:
+  class_property line : Int32?
+
+  # :nodoc:
+  class_property slowest : Int32?
+
+  # :nodoc:
+  class_property? fail_fast = false
+
+  # :nodoc:
+  class_property? focus = false
+
+  # :nodoc:
+  def self.add_location(file, line)
+    locations = @@locations ||= {} of String => Array(Int32)
+    lines = locations[File.expand_path(file)] ||= [] of Int32
+    lines << line
+  end
+
+  # :nodoc:
+  def self.add_tag(tag)
+    if anti_tag = tag.lchop?('~')
+      (@@anti_tags ||= Set(String).new) << anti_tag
+    else
+      (@@tags ||= Set(String).new) << tag
+    end
+  end
+
+  # :nodoc:
+  class_getter randomizer_seed : UInt64?
+  class_getter randomizer : Random::PCG32?
+
+  # :nodoc:
+  def self.order=(mode)
+    seed =
+      case mode
+      when "default"
+        nil
+      when "random"
+        Random::Secure.rand(1..99999).to_u64 # 5 digits or less for simplicity
+      when UInt64
+        mode
+      else
+        raise ArgumentError.new("order must be either 'default', 'random', or a numeric seed value")
+      end
+
+    @@randomizer_seed = seed
+    @@randomizer = seed ? Random::PCG32.new(seed) : nil
+  end
+
+  # :nodoc:
+  class_property option_parser : OptionParser = begin
+    OptionParser.new do |opts|
+      opts.banner = "crystal spec runner"
+      opts.on("-e ", "--example STRING", "run examples whose full nested names include STRING") do |pattern|
+        Spec.pattern = Regex.new(Regex.escape(pattern))
+      end
+      opts.on("-l ", "--line LINE", "run examples whose line matches LINE") do |line|
```

Comment:
Minor, but if we're changing this, can you remove this bonus space after the `-l` (also in `-e` above)?